### PR TITLE
COP-11439 Added an assignees filter to the task pages endpoint

### PR DESCRIPTION
**Pull request by FrancisBry**
Added an "assignees" filter to the /v1/targeting-tasks/pages endpoint. Pass in null or an empty array for all assignees or an array of string for specific assignees.

### CODE REVIEW

File: src/main/java/uk/gov/homeoffice/cop/workflow/targeting/adapter/camunda/tasks/task/TaskFilter.java [hunk 1]
```diff
@@ -32,8 +34,21 @@ private TaskQuery filterMovementModes(TaskQuery query, Collection<MovementMode>
...
+        return query.processVariableValueEquals(HAS_SELECTORS_NAME, toYesNo(hasSelectors.get()));
     }
 
+    private static TaskQuery filterAssignees(TaskQuery query, Collection<String> assignees) {
+        if (CollectionUtils.isEmpty(assignees)) {
+            return query;
+        }
+
+        TaskQuery filterQuery = query.or();
+        assignees.forEach(assignee -> filterQuery.taskAssignee(assignee));
```
Comment (FrancisBry) [inline src/main/java/uk/gov/homeoffice/cop/workflow/targeting/adapter/camunda/tasks/task/TaskFilter.java:51]: This is for new and in-progress tasks so the way to identity who they're assigned to is the assignee

File: src/main/java/uk/gov/homeoffice/cop/workflow/targeting/adapter/camunda/tasks/process/ProcessFilter.java [hunk 2]
```diff
@@ -33,8 +36,21 @@ private ProcessInstanceQuery filterMovementModes(ProcessInstanceQuery query, Col
...
+        return query.variableValueEquals(HAS_SELECTORS_NAME, toYesNo(hasSelectors.get()));
     }
 
+    private static ProcessInstanceQuery filterAssignees(ProcessInstanceQuery query, Collection<String> assignees) {
+        if (CollectionUtils.isEmpty(assignees)) {
+            return query;
+        }
+
+        ProcessInstanceQuery filterQuery = query.or();
+        assignees.forEach(assignee -> filterQuery.variableValueEquals(INITIATED_BY, assignee));
```
Comment (FrancisBry) [inline src/main/java/uk/gov/homeoffice/cop/workflow/targeting/adapter/camunda/tasks/process/ProcessFilter.java:53]: This is for issued tasks so the way to see who it's assigned to is by checking who initiated the issue target form.

File: src/main/java/uk/gov/homeoffice/cop/workflow/targeting/adapter/camunda/tasks/historicprocess/HistoricProcessFilter.java [hunk 3]
```diff
@@ -32,8 +35,21 @@ private HistoricProcessInstanceQuery filterMovementModes(HistoricProcessInstance
...
+        return query.variableValueEquals(HAS_SELECTORS_NAME, toYesNo(hasSelectors.get()));
     }
     
+    private static HistoricProcessInstanceQuery filterAssignees(HistoricProcessInstanceQuery query, Collection<String> assignees) {
+        if (CollectionUtils.isEmpty(assignees)) {
+            return query;
+        }
+
+        HistoricProcessInstanceQuery filterQuery = query.or();
+        assignees.forEach(assignee -> filterQuery.variableValueEquals(INITIATED_BY, assignee));
```
Comment (FrancisBry) [inline src/main/java/uk/gov/homeoffice/cop/workflow/targeting/adapter/camunda/tasks/historicprocess/HistoricProcessFilter.java:52]: This is for completed tasks and like with issued tasks the way to check who it was assigned to is to check who initiated the complete/dismiss/issue target form

**Review by adam-thorpe — APPROVED** [review]

**Comment by michaelruocco** [comment]
LGTM

**Pull request merged**